zebedee-cms/src/test/java/com/github/onsdigital/zebedee/keyring/store/CollectionKeyStoreImplTest.java

```diff
@@ -0,0 +1,269 @@
+package com.github.onsdigital.zebedee.keyring.store;
+
+import com.github.onsdigital.zebedee.keyring.KeyringException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+import static com.github.onsdigital.zebedee.keyring.store.CollectionKeyStoreImpl.COLLECTION_KEY_ALREADY_EXISTS_ERR;
+import static com.github.onsdigital.zebedee.keyring.store.CollectionKeyStoreImpl.COLLECTION_KEY_NOT_FOUND_ERR;
+import static com.github.onsdigital.zebedee.keyring.store.CollectionKeyStoreImpl.COLLECTION_KEY_NULL_ERR;
+import static com.github.onsdigital.zebedee.keyring.store.CollectionKeyStoreImpl.INVALID_COLLECTION_ID_ERR;
+import static com.github.onsdigital.zebedee.keyring.store.CollectionKeyStoreImpl.KEY_DECRYPTION_ERR;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertTrue;
+
+public class CollectionKeyStoreImplTest {
+
+    static final String TEST_COLLECTION_ID = "1234567890";
+    static final String CIPHER_ALGORITHM = "AES/CBC/PKCS5Padding";
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    private CollectionKeyStore store;
+    private File keyringDir;
+
+    @Before
+    public void setUp() throws Exception {
+        keyringDir = folder.newFolder("keyring");
+    }
+
+    @Test(expected = KeyringException.class)
+    public void read_collectionIDNull_shouldThrowException() throws Exception {
+        store = new CollectionKeyStoreImpl(keyringDir.toPath(), null, null);
+
+        SecretKey key = null;
+        try {
+            key = store.read(null);
+        } catch (KeyringException ex) {
+            assertThat(key, is(nullValue()));
+            assertThat(ex.getMessage(), equalTo(INVALID_COLLECTION_ID_ERR));
+            throw ex;
+        }
+    }
+
+    @Test(expected = KeyringException.class)
+    public void read_collectionIDEmpty_shouldThrowException() throws Exception {
+        store = new CollectionKeyStoreImpl(keyringDir.toPath(), null, null);
+
+        SecretKey key = null;
+        try {
+            key = store.read("");
```

Comment:
For the tests that assert on exceptions, the return value is never assigned so is always null. The assertion for `key` could be removed. (same applies for other tests)